ios/CovidShield/CovidShield.m

```diff
@@ -25,28 +27,59 @@ @implementation CovidShield
-      return;
-    }
-    if ([(NSHTTPURLResponse *)response statusCode] != 200) {
-      reject([NSString stringWithFormat:@"%ld", (long)[(NSHTTPURLResponse *)response statusCode]],
-             [NSString stringWithContentsOfURL:location encoding:NSUTF8StringEncoding error:nil], nil);
-      return;
-    }
-    NSURL *temporaryDirectoryURL = [NSURL fileURLWithPath: NSTemporaryDirectory() isDirectory: YES];
-
-    NSURL* destination = [temporaryDirectoryURL URLByAppendingPathComponent: [NSString stringWithFormat:@"%@.zip", [[NSUUID UUID] UUIDString]]];
-
-    [[NSFileManager defaultManager] copyItemAtURL:location toURL:destination error:nil];
-    resolve(destination.path);
-  }] resume];
+  [[self downloadTaskWithURL: [RCTConvert NSURL:url] failureHandler: reject
+              successHandler: ^(NSURL *fileLocation) {
+        NSURL *temporaryDirectoryURL = [NSURL fileURLWithPath: NSTemporaryDirectory() isDirectory: YES];
+
+        NSURL* destination = [temporaryDirectoryURL URLByAppendingPathComponent: [NSString stringWithFormat:@"%@.zip", [[NSUUID UUID] UUIDString]]];
+
+        [[NSFileManager defaultManager] copyItemAtURL:fileLocation toURL:destination error:nil];
+        resolve(destination.path);
+      }] resume];
+}
+
+RCT_REMAP_METHOD(fetchExposureConfiguration, fetchExposureConfigurationWithURL: (NSString *)url withResolver:(RCTPromiseResolveBlock)resolve rejecter:(RCTPromiseRejectBlock)reject)
+{
+  [[self downloadTaskWithURL: [RCTConvert NSURL:url] failureHandler: reject
+              successHandler: ^(NSURL *fileLocation) {
+        resolve([NSString stringWithContentsOfURL: fileLocation encoding: NSUTF8StringEncoding
+                                            error: nil]);
+      }] resume];
+}
+
+#pragma mark Private methods
+
+- (NSURLSessionTask *)downloadTaskWithURL: (NSURL *)url failureHandler: (RCTPromiseRejectBlock)reject successHandler: (void (^)(NSURL*))successHandler {
+  return [[self configuredSessionWithCompletionHandler: ^(NSURL *location, NSURLResponse *resp, NSError *error) {
+        if (error) {
+          reject([NSString stringWithFormat:@"%ld", (long)error.code], error.localizedDescription,
+                 error);
+        } else if ([(NSHTTPURLResponse *)resp statusCode] != 200) {
+          reject([NSString stringWithFormat:@"%ld", (long)[(NSHTTPURLResponse *)resp statusCode]],
+                 [NSString stringWithContentsOfURL: location encoding: NSUTF8StringEncoding
+                                             error: nil], nil);
+        } else {
+          successHandler(location);
+          [[NSFileManager defaultManager] removeItemAtURL:location error:nil];
+        }
+      }] downloadTaskWithURL: url];
+}
+
+- (NSURLSession *)configuredSessionWithCompletionHandler: (void (^)(NSURL *, NSURLResponse *, NSError *)) completion {
+  NSURLSessionConfiguration *sconf = [NSURLSessionConfiguration defaultSessionConfiguration];
```

Comment:
Please change this to use:
`[NSURLSessionConfiguration ephemeralSessionConfiguration]`

This was done in a previous PR, to ensure that downloaded items won't be stored in cache. I'm just pointing this out so it doesn't get lost in the merge.

https://github.com/cds-snc/covid-alert-app/pull/859

Reply:
addressed by f5d49e3c5f50abb94abd7923cbfd084a4355c270